src/rust/engine/process_execution/remote/Cargo.toml

```diff
@@ -37,6 +37,8 @@ strum = { workspace = true }
 strum_macros = { workspace = true }
 parking_lot = { workspace = true }
 opendal = { workspace = true }
+remote_provider = { path = "../../remote_provider" }
+remote_provider_reapi = { path = "../../remote_provider/remote_provider_reapi" }
```

Comment:
This isn't in your graph in the PR description; and it seems maybe a bit leaky if we're just picking up this dependency for `apply_headers`?

Reply:
> seems maybe a bit leaky if we're just picking up this dependency for apply_headers?

Yeah, I guess theoretically this should be entirely an implementation detail of providers surfaced by `remote_provider`... except currently the remote execution isn't wrapped up into a "provider" interface, just hard-coded to Bazel REAPI (and I could imagine it will remain this way), so it felt okay to leak here.

> This isn't in your graph in the PR description

Oops good catch, fixed and updated the text to call it out too.